cmd/podman/images/scp.go

```diff
@@ -0,0 +1,350 @@
+// iden is a string array of ssh identity keys, url is a string array of paths to ssh clients. The first entry
+// in each array is the save location and the second entry is the load location if applicable. An empty string will be
+// inserted if identity is blank (not required in some cases).
+// serv is a map of service destinations with the connection name as the index intended to be used as EngineConfig.ServiceDestinations
+// this function modifies the global scpOpt entities: FromRemote, ToRemote, Connections, and DestImageName
+func parseArgs(args []string, cfg *config.Config) ([]string, []string, map[string]config.Destination, error) {
+	iden := []string{}
+	url := []string{}
+	serv := map[string]config.Destination{}
+	cliConnections := []string{}
+	switch len(args) {
+	case 1:
+		if strings.Contains(args[0], "::") {
+			scpOpts.FromRemote = true
+			cliConnections = append(cliConnections, args[0])
+		} else {
+			scpOpts.DestImageName = args[0]
+			named, err := reference.ParseNormalizedNamed(scpOpts.DestImageName)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			scpOpts.DestImageName = named.String()
+		}
+	case 2:
+		if strings.Contains(args[0], "::") {
+			if !(strings.Contains(args[1], "::")) && len(strings.Split(args[0], "::")) != 2 { // if an image is specified, this mean we are loading from our client
+				cliConnections = append(cliConnections, args[0])
+				scpOpts.ToRemote = true
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if len(strings.Split(args[0], "::")) == 1 || (len(strings.Split(args[0], "::")) == 2 && len(strings.Split(args[1], "::")) == 2) { // is save->load w/ one image name or save-load w/ two image names
+					cliConnections = append(cliConnections, args[0])
+					cliConnections = append(cliConnections, args[1])
+				} else { // else its a load save (order of args) or invalid
+					cliConnections = append(cliConnections, args[1])
+					cliConnections = append(cliConnections, args[0])
+				}
+			} else {
+				return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify host::image image")
+			}
+		} else if strings.Contains(args[1], "::") && len(cliConnections) < 2 { // if we are given image host::
+			scpOpts.DestImageName = args[0]
+			named, err := reference.ParseNormalizedNamed(scpOpts.DestImageName)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			scpOpts.DestImageName = named.String()
+			scpOpts.ToRemote = true
+			cliConnections = append(cliConnections, args[1])
+		} else {
+			return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify two local images")
+		}
+	default:
+		return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "expected at least 1 argument and no more than 2 arguments")
+	}
+	for i, val := range cliConnections {
+		splitEnv := strings.SplitN(val, "::", 2)
+		scpOpts.Connections = append(scpOpts.Connections, splitEnv[0])
```

Comment:
(From https://github.com/containers/podman/pull/10828#discussion_r672659759 )

Nothing uses `scpOpts.Connections` outside this loop; there can be just a `connectionName := splitEnv[0]` variable local to this loop.

Reply:
I now use `scpOpts.Connections` within the `createConnection` func, I am pushing a bunch of new changes including this in a few minutes